src/memex/models/document.py

```diff
@@ -393,29 +395,35 @@ def merge_documents(session, documents, updated=None):
             m.document = master
             m.updated = updated
 
-        session.delete(doc)
-
     try:
         session.flush()
+        session.query(Annotation) \
+            .filter(Annotation.document_id.in_(duplicate_ids)) \
+            .update({Annotation.document_id: master.id}, synchronize_session='fetch')
```

Comment:
There is one now: https://github.com/hypothesis/h/pull/3930/files#diff-d0e476880deffcb29098651d7d4c5fc3R629
Thanks for spotting this!